### PR TITLE
added quicksort

### PR DESCRIPTION
**Pull request by spamegg1**
- added `.gitignore`
- I tried to follow [OCaml style guide](https://www.seas.upenn.edu/~cis341/current/programming_style.shtml) as much as possible. I am new to OCaml.
- included a `main` to test the function.
- wrote a `partition` function from scratch for educational purposes, instead of using the built-in `List.partition`.

Please let me know if you'd like any changes or suggest improvements.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+### IntelliJ
+.idea/
```
Comment (dmbaturin) [inline .gitignore:2]: Why?
A useful .gitignore could exclude compiles OCaml files: `*.cmo *.cmi *.cmx *.cmxz`.
Feel free to steal mine: https://github.com/dmbaturin/soupault/blob/master/.gitignore

File: Sorts/quicksort.ml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    in
+
+    print_string "Unsorted: ";
+    List.iter (Printf.printf "%d ") list_to_sort;
+    print_newline ();
+
+    print_string "  Sorted: ";
+    List.iter (Printf.printf "%d ") (quicksort list_to_sort);
+    print_newline ();
+    ;;
```
Comment (dmbaturin) [inline Sorts/quicksort.ml:36]: Use of `;;` in source code files (as opposed to the REPL) is discouraged. You can replace the last line with `let () = main ()`.
Matching on unit (`()`) has an additional advantage: if the type of the function is _not_ unit, the compiler will through a type error.
If you intentionally want to ignore the return value no matter the type, you can match on wildcard instead: `let _ = main ()`.

**Review by dmbaturin — COMMENTED** [review]
The PR is good in general but can be improved.

File: .gitignore (hunk 1 above)
Comment (spamegg1) [reply .gitignore:2]: Will do, thanks! I'm new to all this.

File: Sorts/quicksort.ml (hunk 2 above)
Comment (spamegg1) [reply Sorts/quicksort.ml:36]: Got it. Thanks again!

**Pull request merged**